gossipsub.go

```diff
@@ -35,18 +29,31 @@ var (
 	GossipSubFanoutTTL = 60 * time.Second
 )
 
-// NewGossipSub returns a new PubSub object using GossipSubRouter as the router.
-func NewGossipSub(ctx context.Context, h host.Host, opts ...Option) (*PubSub, error) {
-	rt := &GossipSubRouter{
-		peers:   make(map[peer.ID]protocol.ID),
-		mesh:    make(map[string]map[peer.ID]struct{}),
-		fanout:  make(map[string]map[peer.ID]struct{}),
-		lastpub: make(map[string]int64),
-		gossip:  make(map[peer.ID][]*pb.ControlIHave),
-		control: make(map[peer.ID]*pb.ControlMessage),
-		mcache:  NewMessageCache(GossipSubHistoryGossip, GossipSubHistoryLength),
+func NewGossipConfigurableRouter(configs ...GossipConfiguration) *GossipConfigurableRouter {
```

Comment:
don't remove the old constructor.

Reply:
see above, it was moved not *re*moved

Reply:
keep it here and make a new constructor `NewGossipSubRouterWithConfig`

Reply:
sure